File: ClimbingStairs.js
const climbStairs = (n) => {
    
    if (n <= 0){
        return 0;
    }
    if (n == 1){
        return 1;
    }
    if (n == 2){
        return 2;
    }
    
    let s1 = 1;
    let s2 = 2;
    let  result = 0;
    for(let i = 2; i < n; i++){
        result = s1 + s2;
        s1 = s2;
        s2 = result;
    }
    return result;
};


const testClimbStairs = () => {

    const testCases = [
        { input: 1, expected: 1 },
        { input: 2, expected: 2 },
        { input: 3, expected: 3 },
        { input: 4, expected: 5 },
        { input: 5, expected: 8 },
        { input: 6, expected: 13 },
        { input: 7, expected: 21 },
    ];

    for (let i = 0; i < testCases.length; i++) {
        const { input, expected } = testCases[i];
        const result = climbStairs(input);

        if (result === expected) {
            console.log(`Test case ${i + 1} passed!`);
        } else {
            console.log(`Test case ${i + 1} failed. Expected ${expected}, but got ${result}.`);
        }
    }
};

testClimbStairs();

File: ClimbingStairs.java
import java.util.*;


public class ClimbingStairs {
    public static int climbStairs(int n) {
        if (n <= 0) {
            return 0;
        }
        if (n == 1) {
            return 1;
        }
        if (n == 2) {
            return 2;
        }
        
        int s1 = 1;
        int s2 = 2;
        int result = 0;
        
        for (int i = 2; i < n; i++) {
            result = s1 + s2;
            s1 = s2;
            s2 = result;
        }
        
        return result;
    }
    
    public static void testClimbStairs() {
        List<Map<String, Integer>> testCases = new ArrayList<>();
        testCases.add(createTestCase(1, 1));
        testCases.add(createTestCase(2, 2));
        testCases.add(createTestCase(3, 3));
        testCases.add(createTestCase(4, 5));
        testCases.add(createTestCase(5, 8));
        testCases.add(createTestCase(6, 13));
        testCases.add(createTestCase(7, 21));
        
        for (int i = 0; i < testCases.size(); i++) {
            Map<String, Integer> testCase = testCases.get(i);
            int input = testCase.get("input");
            int expected = testCase.get("expected");
            int result = climbStairs(input);
            
            if (result == expected) {
                System.out.println("Test case " + (i + 1) + " passed!");
            } else {
                System.out.println("Test case " + (i + 1) + " failed. Expected " + expected + ", but got " + result + ".");
            }
        }
    }
    
    public static Map<String, Integer> createTestCase(int input, int expected) {
        Map<String, Integer> testCase = new HashMap<>();
        testCase.put("input", input);
        testCase.put("expected", expected);
        return testCase;
    }
    
    public static void main(String[] args) {
        testClimbStairs();
    }
}
